internal/execution/target/target.go

```diff
@@ -12,12 +15,50 @@ const (
 	TargetTypeAsync
 )
 
-type Target interface {
-	GetExecutionID() string
-	GetTargetID() string
-	IsInterruptOnError() bool
-	GetEndpoint() string
-	GetTargetType() TargetType
-	GetTimeout() time.Duration
-	GetSigningKey() string
+type Target struct {
+	ExecutionID      string              `json:"execution_id"`
+	TargetID         string              `json:"target_id"`
+	TargetType       TargetType          `json:"target_type"`
+	Endpoint         string              `json:"endpoint"`
+	Timeout          time.Duration       `json:"timeout"`
+	InterruptOnError bool                `json:"interrupt_on_error"`
+	SigningKey       *crypto.CryptoValue `json:"signing_key"`
+	SigningKeyDec    string
```

Comment:
Do not export the decrrypted field as it will be marshalled.

Reply:
Had the exact same thought, hence my comment below.